ChangeLog.md

```diff
@@ -1,3 +1,18 @@
+### 0.4.0
+
+#### chruby.sh
+
+* Use the installed Ruby directory name for setting `GEM_HOME`. (@eregon)  
+  This guarantees a unique `GEM_HOME` per installed Ruby, even if they have
+  the same `RUBY_VERSION`.  This is important for native extensions, which
+  might compile differently based on build-time flags such as `--enable-shared`.
+  This also fixes the bug that the `GEM_HOME` for non-MRI Ruby implementations
+  was shared even for different releases.
+
+  In practice, updating `chruby` means you will need to install gems again as
+  the gem directories will be different. It is also recommended to clean old
+  gem directories which will no longer by used with `rm -rf ~/.gem`.
```

Comment:
`~/.gem/` may also contain non-chruby gems that the user might not want deleted.

Reply:
For gems installed with `gem install --user` with the system ruby (`chruby system`)?
I'm not sure what we could recommend instead, not cleaning `.gem` might lead to a lot of wasted disk space.